docs/dev/guidelines/server.rst

```diff
@@ -418,6 +418,26 @@ https://www.baeldung.com/spring-tests
 
 If you want to write tests for Programming Exercises to test student's submissions check out `this <https://confluence.ase.in.tum.de/display/ArTEMiS/Best+Practices+for+writing+Java+Programming+Exercise+Tests+in+Artemis>`__.
 
+23. Counting database query calls within tests
+==============================================
+
+It's possible to write tests that check how many database calls are performed during a REST call. This is useful to ensure that code changes don't lead to more database calls,
+or at least to remind developers in case they do. It's especially important for commonly used endpoints that users access multiple times or every time they use Artemis.
+However, we should consider carefully before adding such assertions to a test as it makes the test more tedious to maintain.
+
+An example on how to track how many database calls are performed during a REST call is shown below. It uses the ``HibernateQueryInterceptor`` which counts the number of queries.
+For ease of use, a custom assert ``assertThatDb`` was added that allows to do the check in one line.
+
+.. code-block:: java
+
+    public class TestClass {
+
+        @Test
+        @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+        public void testQueryCount() throws Exception {
+            assertThatDb(() -> request.get("/api/courses/" + courses.get(0).getId() + "/for-dashboard", HttpStatus.OK, Course.class)).hasBeenCalledTimes(3);
+        }
+    }
```

Comment:
```suggestion
    }

The ``hasBeenCalledTimes(…)`` assertion also returns the result of the request so that further assertions on that are possible.
```

Maybe mention that it is easy to add further assertions on the request result as that is not shown neither here nor in the places where it is already used in the server code.

Reply:
Added to the docs, thanks 👍